android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/av/SharedCookiesDataSourceFactoryProvider.kt

```diff
@@ -0,0 +1,28 @@
+package versioned.host.exp.exponent.modules.universal.av
+
+import android.content.Context
+import com.facebook.react.bridge.ReactContext
+import com.google.android.exoplayer2.upstream.DataSource
+import com.google.android.exoplayer2.upstream.TransferListener
+import expo.modules.av.player.datasource.SharedCookiesDataSourceFactoryProvider
+import expo.modules.core.ModuleRegistry
+import host.exp.exponent.utils.ScopedContext
+
+class SharedCookiesDataSourceFactoryProvider : SharedCookiesDataSourceFactoryProvider() {
+  override fun createFactory(
+    context: Context,
+    moduleRegistry: ModuleRegistry,
+    userAgent: String,
+    requestHeaders: Map<String, Any>?,
+    transferListener: TransferListener
+  ): DataSource.Factory {
+    val reactContext: ReactContext? = if (context is ReactContext) {
+      context
+    } else if (context is ScopedContext) {
+      context.context as ReactContext
+    } else {
+      null
```

Comment:
We are going to throw a NPE when constructing SharedCookiesDataSourceFactory. Better to throw a more informative error here IMO.